### ISSUE TITLE
FATAL ERROR: Unexpectedly found nil

### ISSUE THREAD

**Issue by helloFarrukh** [pusher/pusher-channels-flutter#100]
Hi, 
Please see that it works fine on android but keeps crashing on IOS with the below mentioned error log:

ERROR ON SIMULATOR
`pusher_channels_flutter/SwiftPusherChannelsFlutterPlugin.swift:237: Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value`

ERROR ON REAL DEVICE
```
pusher_channels_flutter/SwiftPusherChannelsFlutterPlugin.swift:237: Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value
pusher_channels_flutter/SwiftPusherChannelsFlutterPlugin.swift:237: Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value
* thread #1, queue = 'com.apple.main-thread', stop reason = Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value
    frame #0: 0x0000000185203ebc libswiftCore.dylib`_swift_runtime_on_report
libswiftCore.dylib`:
->  0x185203ebc <+0>: ret    
libswiftCore.dylib`:
    0x185203ec0 <+0>: b      0x185203ebc               ; _swift_runtime_on_report
libswiftCore.dylib`:
    0x185203ec4 <+0>: adrp   x8, 464674
    0x185203ec8 <+4>: ldrb   w0, [x8, #0x8c]
Target 0: (Runner) stopped.
Lost connection to device.
```

I am using `pusher_channels_flutter: ^2.1.2`

**Flutter Doctor**
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.0, on macOS 13.1 22C65 darwin-arm64, locale en-PK)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.1)
[✓] Connected device (3 available)
[!] HTTP Host Availability
    ✗ HTTP host "https://cocoapods.org/" is not reachable. Reason: Failed to connect to host in 10 seconds

! Doctor found issues in 1 category.

**Comment by gabrielginter** [comment]
I was experiencing the same issue, in my case it was happening by calling `PusherChannelsFlutter.disconnect()` on a non initialized pusher instance, i.e. before calling `PusherChannelsFlutter.init`. Perhaps a small fix to prevent the crash when calling `disconnect` by accident should be implemented.

**Comment by benjamin-tang-pusher** [comment]
@gabrielginter I couldn't reproduce this by calling disconnect() before init(), although I did initialize it with getInstance(). Did you do that in your app?

**Comment by gabrielginter** [comment]
@benjamin-tang-pusher I used `getInstance()` before, it is how I get the `PusherChannelsFlutter` instance.

To clarify, this was happening to me very early in the app's lifecycle, my app was logging out a banned user on startup, so on logout I terminate pusher (calling `disconnect()`), perhaps it is the fact that I was calling this method `disconnect` milliseconds after calling `getInstance()`. After I added a flag in my Pusher wrapper class to only call 'disconnect' if `init` had ran successfully, the issue went away.

I hope it helps.
